src/Openbuildings/Spiderling/Driver/Phantomjs/Connection.php

```diff
@@ -233,6 +239,18 @@ public function command_url($command)
 	}
 
 	/**
+	 * Set phantomjs binary location
+	 * @param string $binary
+	 * @throws Exception
+	 */
+	public function set_phantomjs_binary($binary) {
+			if ( ! is_file($binary)){
```

Comment:
Also you cannot check if the string being set here is valid unless you try to execute it. It could be an alias, shell function etc. So the `is_file()` check would create false negatives in many cases. You can instead try to check the exit code when executing it, but I'm not sure if that would be easy either as the process doesn't exit unless killed.